src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RedirectedRuntimeBasicWriter.cs

```diff
@@ -56,20 +57,31 @@ public override void WriteHtmlContent(CSharpRenderingContext context, HtmlConten
         {
             const int MaxStringLiteralLength = 1024;
 
+            var builder = new StringBuilder();
+            for (var i = 0; i < node.Children.Count; i++)
+            {
+                if (node.Children[i] is RazorIRToken token && token.IsHtml)
+                {
+                    builder.Append(token.Content);
+                }
+            }
+
+            var content = builder.ToString();
```

Comment:
If I can do a suggestion, A solution can be to create only one `new StringBuilder(MaxStringLiteralLength)`, and fill it token by token until the limit is reached. Then a `StringBuilder.Clear()` will reset the StringBuilder without destroying its internal buffer.
It will avoid doing this huge string allocation. But it would make small string allocations necessary.